pkg/apis/core/validation/validation.go

```diff
@@ -6135,6 +6227,39 @@ func validateWindowsHostProcessPod(podSpec *core.PodSpec, fieldPath *field.Path,
 	return allErrs
 }
 
+// validateOS validates the OS field within pod spec
+func validateOS(podSpec *core.PodSpec, fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// These valid values currently are linux, windows.
+	// In future, they can be expanded to values from
+	// https://github.com/opencontainers/runtime-spec/blob/master/config.md#platform-specific-configuration
+	validOS := sets.NewString(string(core.Linux), string(core.Windows))
+	os := podSpec.OS
+	if os == nil {
+		return allErrs
+	}
+	if len(os.Name) == 0 {
+		return allErrs
+	}
+	osName := strings.ToLower(string(os.Name))
+	for _, msg := range validation.IsDNS1123Subdomain(osName) {
+		allErrs = append(allErrs, field.Invalid(fldPath, osName, msg))
+	}
+	if !validOS.Has(osName) {
+		errMsg := fmt.Sprintf("The os field should have a valid value and it should be one of %v",
+			strings.Join(validOS.List(), ","))
+		allErrs = append(allErrs, field.Invalid(fldPath, osName, errMsg))
+	}
+	// Fail Pod validation if feature is not enabled (unless podspec already exists and contains OS fields) instead of dropping fields
+	// TODO: Check if this is needed
+	if !opts.AllowOSField {
```

Comment:
this is ok as a backstop, but we should also conditionally drop the os field in PrepareForCreate/PrepareForUpdate as per https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-field-in-existing-api-version